library/src/main/java/com/google/android/exoplayer/hls/Variant.java

```diff
@@ -27,10 +27,15 @@
   public final String url;
   public final Format format;
 
-  public Variant(int index, String url, int bitrate, String codecs, int width, int height) {
+  public Variant(int index, String name, String url, int bitrate, String codecs, int width, int height) {
     this.url = url;
-    format = new Format(Integer.toString(index), MimeTypes.APPLICATION_M3U8, width, height, -1, -1,
-        -1, bitrate, null, codecs);
+    if (name != null) {
```

Comment:
Hm. I guess it would be a weird case (some variants with names, others without in the master playlist), but it would be strange if we ended up exposing some tracks with indices for names and other tracks with actual names. How would you feel about:

No name: Integer.toString(index)
With name: Integer.toString(index) + ':' + name

The same will apply for the TODO in the WebmExtractor, where I would guess some tracks having names and others not having them may be more likely.